File: BACKEND-APPLICATION/src/main/java/com/application/entities/Hotel.java
package com.application.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "hotels")
@Setter
@Getter
public class Hotel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String description;
    private String imageUrl;
    private boolean isAvailable;
    private boolean isApproved;
    @ManyToOne
    private Owner owner;
    @ManyToOne
    private City city;
    @OneToMany(mappedBy = "hotel")
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private List<Room> rooms = new ArrayList<>();

    public Hotel() { }

}

File: FRONTEND-APPLICATION/src/app/models/hotel.ts
import { User } from "./user";

export class Hotel {

  public id: number | undefined;
  public name: string;
  public description: string;
  public imageUrl: string;
  public owner: User;
  public available: boolean;
  public approved: boolean;

  constructor() {
    this.id = undefined;
    this.name = '';
    this.description = '';
    this.imageUrl = '';
    this.owner = new User();
    this.available = false;
    this.approved = false;
  }

}